Stack/Opc.Ua.Core/Stack/Server/ServerBase.cs

```diff
@@ -1152,7 +1152,7 @@ private static string GetBestDiscoveryUrl(Uri clientUrl, BaseAddress baseAddress
                         continue;
                     }
 
-                    if (endpointUrl.Port != baseAddress.Url.Port)
+                    if (endpointUrl.Port != baseAddress.Url.Port && !translateHttpsEndpoint)
```

Comment:
@BoBiene If this check is added than you will return duplicate HttpsBinaryTransport endpoints in case your configuration contains two such base addresses which differ only through their port numbers.

probably a better solution in your case is to completely remove this if clause and add the one bellow